### ISSUE TITLE
Site Setup List triggering multiple WCAG rule violations as a tablist

### ISSUE THREAD

**Issue by youbrokesomething** [Automattic/wp-calypso#75588]
### Quick summary

On /home, the site setup list (.site-setup-list__list) is currently an unordered list with focusable buttons nested in list items and the roles tablist and tab. Accessibility rule checkers hate this because there are nested focusable elements and because the list items don't have the aria roles.

There are three ways to fix this:

1. Because a tablist is being used, the [tablist pattern with automatic focus](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/) should be used instead.
2. Drop the tablist and tab roles and make the list items clickable instead of nesting a button.
3. Leave everything as is and just give the list items the role "presentation" which will allow assistive technology and tools to skip right to the buttons.

### Steps to reproduce

1. Open any simple site to /Home.
2. Run an accessibility checking tool (like axe DevTools Chrome extension) or inspect the site setup list.

### What you expected to happen

Accessibility rule-checking tools shouldn't find any errors. Focusable elements shouldn't be nested.

### What actually happened

Six list item elements spawned thirteen accessibility rule violations.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by autumnfjeld** [comment]
I believe we can close this because the LaunchPad makes this list obsolete.  Is that correct @Addison-Stavlo ? Or do some users still interact with the My Home site setup list.

LaunchPad info:
p58i-cVq-p2
p7DVsv-gje-p2

**Comment by dsas** [comment]
Implementing the [tablist pattern with automatic focus](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/) will basically mean re-writing the whole thing. 

The site setup list is not obsolete yet and contains these items:
<img width="360" alt="Screenshot on 2023-05-17 at 08-46-36" src="https://github.com/Automattic/wp-calypso/assets/93301/18d61f17-3aff-4681-97b1-35c2351b4d63">

That said it looks decreasingly relevant due to Launchpad, and could be obsoleted in the near future: p9Jlb4-7c4-p2

Option three will be a quick change that will remove the a11y linter warning while the focus behaviour will continue to work as-is. I'll open a PR for that.
